data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/LengthExpressionFunction.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import org.opensearch.dataprepper.model.event.Event;
+import java.util.List;
+import javax.inject.Named;
+import java.util.function.Function;
+
+@Named
+public class LengthExpressionFunction implements ExpressionFunction {
+    public String getFunctionName() {
+        return "length";
+    }
+
+    public Object evaluate(final List<Object> args, Event event, Function<Object, Object> convertLiteralType) {
+        if (args.size() != 1) {
+            throw new RuntimeException("length() takes only one argument");
+        }
+        Object arg = args.get(0);
+        if (!(arg instanceof String)) {
+            throw new RuntimeException("length() takes only String type arguments");
+        }
+        String argStr = ((String)arg).trim();
+        if (argStr.length() == 0) {
+            return 0;
+        }
+        if (argStr.charAt(0) == '\"') {
```

Comment:
Is this something we should support?

I'm not sure it adds a lot of value. And it could lead to confusing situations for users.

Reply:
I tend to think we should start simply by assuming that the first (and only) argument must be a JSON Pointer. If it is not, the expression should fail.